src/components/users_cache.rs

```diff
@@ -0,0 +1,125 @@
+use deadpool_redis::redis::{cmd, RedisResult};
+
+use crate::utils::encrypt_string::hash_with_key;
+
+use super::redis::RedisComponent;
+
+pub struct UsersCacheComponent<T: RedisComponent> {
+    redis_component: T,
+    hashing_key: String,
+}
+
+impl<T: RedisComponent> UsersCacheComponent<T> {
+    pub fn new(redis: T, hashing_key: String) -> Self {
+        Self {
+            redis_component: redis,
+            hashing_key,
+        }
+    }
+
+    pub async fn add_user(&mut self, token: String, user_id: String) -> Result<(), String> {
+        let con = self.redis_component.get_async_connection().await;
+
+        if con.is_none() {
+            let error = format!(
+                "Couldn't cache user {}, redis has no connection available",
+                user_id
+            );
+            log::error!("{}", error);
+            return Err(error);
+        }
+
+        let key = hash_with_key(token.clone(), self.hashing_key.clone());
+
+        let mut connection = con.unwrap();
+        let res = cmd("SET")
```

Comment:
How much can this grow? I think we need to set the TTL for the `SET` here.

Reply:
The default expiring time is now 2 minutes (around 2 calls to sync request)